src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java

```diff
@@ -86,17 +89,20 @@
       return Arrays.<GapicProvider<? extends Object>>asList(provider);
 
     } else if (id.equals(CSHARP)) {
-      GapicProvider<? extends Object> provider =
-          CommonGapicProvider.<Interface>newBuilder()
+      GapicCodePathMapper pathMapper =
+          CommonGapicCodePathMapper.newBuilder()
+              .setPrefix("")
+              .setShouldAppendPackage(true)
+              .setPackageFilePathNameFormatter(new CSharpNameFormatter())
+              .build();
+      GapicProvider<? extends Object> mainProvider =
+          ViewModelGapicProvider.newBuilder()
               .setModel(model)
-              .setView(new InterfaceView())
-              .setContext(new CSharpGapicContext(model, apiConfig))
```

Comment:
Could you delete CSharpGapicContext and other orphaned classes as part of this PR?